module/VuFind/src/VuFind/View/Helper/Root/CleanHtmlFactory.php

```diff
@@ -134,23 +149,25 @@ protected function createPurifier(array $options): HTMLPurifier
      */
     protected function setAdditionalConfiguration(HTMLPurifier_Config $config, array $options): void
     {
+        // Configure allowed elements:
         $context = $options['context'] ?? 'default';
-        if ('heading' === $context) {
-            // Limit elements allowed in headings
-            // (see https://developer.mozilla.org/en-US/docs/Web/HTML/Content_categories#phrasing_content for more
-            // information):
-            $config->set('HTML.AllowedElements', 'a,b,br,em,i,span');
-        } else {
-            // Add support for details and summary elements:
-            $definition = $config->getHTMLDefinition(true);
-            $definition->addElement(
-                'details',
-                'Block',
-                'Flow',
-                'Common',
-                ['open' => new \HTMLPurifier_AttrDef_HTML_Bool(true)]
-            );
-            $definition->addElement('summary', 'Block', 'Flow', 'Common');
+        $contextConfig
+            = (array)($this->container->get(ConfigPluginManager::class)->get('config')->HTML_Rendering_Contexts ?? []);
+        $allowedElements
+            = $contextConfig['allowed_elements'][$context] ?? $this->defaultContextConfig[$context] ?? null;
```

Comment:
I think it might be more efficient to process the configuration once and modify the `defaultContextConfig` property directly, instead of potentially processing it many times on demand. But maybe I'm misunderstanding the flow here.

Reply:
True. Now done.